spec/controllers/comments_controller_spec.rb

```diff
@@ -446,6 +446,32 @@
         end
       end
     end
+
+    context "guest comments are turned on in admin settings" do
+      let(:work) { create(:work) }
+
+      it "allows guest comments" do
+        post :create, params: { work_id: work.id, comment: anon_comment_attributes }
```

Comment:
I think there are some other things we need to be testing here:
* the other actions affected by `check_guest_comment_permissions` (so `new` and `add_comment_reply`)
* how this interacts with other settings on commentables (i.e., when the creator of a work disallows all comments) -- I'm okay with just checking works in this case, although someone else might have a good argument for also testing admin posts
* that logged-in users can still comment (it's probably okay to just test this for when guest comments are off)

Reply:
So these shouldn't all be bunched here under `describe "POST #create"`: the `add_comment_reply` tests should be in the `describe "GET #add_comment_reply"` block and the `new` tests in `describe "POST #new"`.